### ISSUE TITLE
cron does not work on recent alpine 3.4 templates (both LXC and LXD).

### ISSUE THREAD

**Issue by melato** [lxc/lxc#1164]
crontab does not work on recent LXD alpine 3.4 images.  Also happens with LXC.

To reproduce:

```
lxc launch images:alpine/3.4/amd64 alpine34
lxc exec alpine34 sh

crontab -e
```

add the following line:
```
*	*	*	*	*	/bin/date > /root/date.log
```

wait a minute
ls /root

Result:  There is nothing there.
Expected Result:  There should be a file /root/date.log

I have tracked this down and found the problem and a fix:
It is caused by a broken symlink:
/etc/runlevels/default/cron -> /etc/init.d/cron
It should probably be:
/etc/runlevels/default/crond -> /etc/init.d/crond

If I replace the symlink and reboot, cron works.

The symlink is created by lxc-alpine.in, in setup_services():
```
for svc_name in networking cron; do
  ln -s /etc/init.d/$svc_name etc/runlevels/default/$svc_name
done

```
I suppose changing "cron" to "crond" in lxc-alpine.in will fix this bug.

Apparently "cron" was renamed to "crond" in a recent alpine package update.

**Comment by melato** [comment]
But fixing lxc-alpine.in this way may break alpine 3.0 images (and possibly other versions).  The alpine 3.0 LXD image has not been updated since June 30, 2016, and it does not have this problem.

**Comment by stgraber** [comment]
Alpine 3.0 is end of life, I stopped building them a little while back but forgot to wipe them from the image server, will do that now.

**Comment by melato** [comment]
I should add that when editing crontab, you should add a blank line at the end.  I learned the hard way that busybox cron may fail to read the last line if it is not terminated with a newline.  I had tested all alpine versions, but I would need to redo the test to find out which images have the problem.  Checking for a broken /etc/runlevels/default/cron link is a faster way to check if the image is affected.

**Comment by stgraber** [comment]
Can you send a branch to fix the alpine template?

**Comment by melato** [comment]
I can write and test the fix, but I"m not sure how to "send a branch".  It looks like I need to follow the [Creating a pull request from a fork](https://help.github.com/articles/creating-a-pull-request-from-a-fork/) procedure, so fork, branch, commit, and create pull request.  I'll do that.

**Comment by melato** [comment]
I created a pull request with the fix:  https://github.com/lxc/lxc/pull/1172

**Comment by brauner** [comment]
Should be closed by #1173. Thanks @melato.
